hatch_pip_compile/plugin.py

```diff
@@ -92,17 +101,20 @@ def _hatch_pip_compile_install(self):
             )
             if not self.lockfile_up_to_date:
                 self._pip_compile_cli()
-            self._pip_sync_cli()
-        if not self.skip_install:
-            if self.dev_mode:
-                super().install_project_dev_mode()
-            else:
-                super().install_project()
+            self._piptools_install_dependencies()
+            if not self.skip_install:
+                if self.dev_mode:
+                    self.install_project_no_deps_dev()
+                else:
+                    self.install_project_no_deps()
 
     def _pip_compile_cli(self) -> None:
         """
         Run pip-compile
         """
+        if not self.dependencies and self._piptools_lock_file.exists():
```

Comment:
Hmm what happens <code>if not self.dependencies and <b>not</b> self._piptools_lock_file.exists():</code>

Reply:
https://github.com/juftin/hatch-pip-compile/blob/a837217146fb7fd8ce9217910e9c03fb2b336b79/hatch_pip_compile/plugin.py#L237-L238

Then `dependencies_in_sync = true` and `_pip_compile_cli` doesn't get called. 

---

But when `not self.dependencies and self._piptools_lock_file.exists()` then `dependencies_in_sync = false` and `_pip_compile_cli` will delete the lockfile.